src/framework/mpas_attlist.F

```diff
@@ -246,13 +254,259 @@ subroutine mpas_add_att_text(attList, attName, attValue, ierr)!{{{
          allocate(cursor % next)
          cursor => cursor % next
       end if
-     
+
       cursor % attType = MPAS_ATT_TEXT
       write(cursor % attName,'(a)') trim(attName)
       write(cursor % attValueText,'(a)') trim(attValue)
 
    end subroutine mpas_add_att_text!}}}
 
+!***********************************************************************
+!
+! routine mpas_modify_att_text
+!
+! > \brief  MPAS modify text attribute routine
+! > \author Matthew Dimond
+! > \date   06/27/23
+! > \details
+! > This routine modifies a text attribute in the attribute list,
+! > and returns a 1 in ierr if the attribute is not found, or the attribute
+! > has a type incompatible with attValue.
+!
+!----------------------------------------------------------------------
+  subroutine mpas_modify_att_text(attList, attName, attValue, ierr)!{{{
+
+    implicit none
+
+    type (att_list_type), pointer :: attList !< Input/Output: Attribute List
+    character (len=*), intent(in) :: attName !< Input: Att. name to modify
+    character (len=*), intent(in) :: attValue !< Input: Updated Att. value
+    integer, intent(out), optional :: ierr !< Output: Error flag
+
+    type (att_list_type), pointer :: cursor
+
+    if (present(ierr)) ierr = 0
+
+    ! Return an error if there is somehow no attList
+    if (.not. associated(attList)) then
+      ierr = 1
```

Comment:
Since `ierr` is an optional argument, we should verify that it's `present` before setting it here.